### PR TITLE
validate: add root.path validation when platform is windows

### PR DESCRIPTION
**Pull request by q384566678**
According to the [spec.](https://github.com/opencontainers/runtime-spec/blame/master/config.md#L37)

>On Windows, path MUST be a volume GUID path.

Signed-off-by: zhouhao <zhouhao@cn.fujitsu.com>

### CODE REVIEW

File: validate/validate.go [hunk 1]
```diff
@@ -219,6 +219,11 @@ func (v *Validator) CheckRoot() (errs error) {
 	}
 
 	if v.platform == "windows" {
+		if !strings.HasPrefix(v.spec.Root.Path, `\\?\`) {
```
Comment (wking) [inline validate/validate.go]: We can be stricter than this.  The spec links [here][1], whivh would be a regex like `[\][\][?][\]Volume[
{][a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}[}][\]?"`.  The Windows docs say "GUID", I think they mean "[UUID][2]", based on their `\\?\Volume{26a21bda-a627-11d7-9931-806e6f6e6963}\` example.

Also, unit tests :).

[1]: https://msdn.microsoft.com/en-us/library/windows/desktop/aa365248(v=vs.85).aspx
[2]: https://tools.ietf.org/html/rfc4122#section-3

Comment (q384566678) [reply validate/validate.go]: updated, PTAL.

File: validate/validate_test.go [hunk 2]
```diff
@@ -114,6 +114,7 @@ func TestCheckRoot(t *testing.T) {
 	}{
 		{rspec.Spec{Windows: &rspec.Windows{HyperV: &rspec.WindowsHyperV{}}, Root: &rspec.Root{}}, "windows", specerror.RootOnHyperV},
 		{rspec.Spec{Windows: &rspec.Windows{HyperV: &rspec.WindowsHyperV{}}, Root: nil}, "windows", specerror.NonError},
+		{rspec.Spec{Windows: &rspec.Windows{}, Root: &rspec.Root{Path: filepath.Join(tmpBundle, "rootfs")}}, "windows", specerror.PathFormatOnWindows},
```
Comment (wking) [inline validate/validate_test.go]: Can we also check that [this][1] passes validation?  The regex is large, and a positive match would help convince me that it wasn't broken.

[1]: https://github.com/opencontainers/runtime-spec/blob/v1.0.0/config.md#example-windows

Comment (q384566678) [reply validate/validate_test.go]: updated.

File: generate/generate.go [hunk 3]
```diff
@@ -1089,11 +1089,7 @@ func (g *Generator) DropProcessCapability(c string) error {
 		}
 	}
 
-	if err := validate.CapValid(cp, false); err != nil {
-		return err
-	}
-
-	return nil
+	return validate.CapValid(cp, false)
```
Comment (wking) [inline generate/generate.go]: I like this change, but it seems orthogonal.  Can 1a9532e get its own PR?

**Comment by Mashimiao** [comment]
LGTM

**Comment by q384566678** [comment]
ping @mrunalp @liangchenye

**Comment by liangchenye** [comment]
LGTM
